controllers/dscinitialization/servicemesh_setup.go

```diff
@@ -205,12 +205,6 @@ func (r *DSCInitializationReconciler) authorizationFeatures(instance *dsciv1.DSC
 				).
 				PostConditions(
 					feature.WaitForPodsToBeReady(serviceMeshSpec.ControlPlane.Namespace),
-				).
-				OnDelete(
-					servicemesh.RemoveExtensionProvider(
```

Comment:
`servicemesh.RemoveExtensionProvider` becomes dead  code - why not remove it as well?